### PR TITLE
Fix folder IDs to comply with api spec

### PR DESCRIPTION
**Pull request by mvn23**
According to the subsonic [api specification](http://www.subsonic.org/pages/inc/api/schema/subsonic-rest-api-1.9.0.xsd) (see snippet below), MusicFolders should have an integer id. Currently, this is implemented in supysonic with an uuid.
This PR fixes that. All folders now have an integer ID. This also fixes the issue where the music folder dropdown in Jamstash does not work.

```
<xs:complexType name="MusicFolder">
  <xs:attribute name="id" type="xs:int" use="required"/>
  <xs:attribute name="name" type="xs:string" use="optional"/>
</xs:complexType>
```

### CODE REVIEW

File: supysonic/api/__init__.py [hunk 1]
```diff
@@ -82,11 +82,27 @@ def get_client_prefs():
...
+    """Return the entity ID as its proper type."""
+    if cls == Folder:
+        try:
+            return int(eid)
+        except ValueError:
+            raise GenericError("Invalid ID")
+    try:
+        return uuid.UUID(eid)
+    except (AttributeError, ValueError):
+        raise GenericError("Invalid ID")
```
[the remark this answers is not in the document]
Comment (mvn23) [reply supysonic/api/__init__.py]: Could be moved to calling code, however this way we handle the invalid id scenario in one place and let Flask further handle the error.

File: supysonic/api/albums_songs.py [hunk 2]
```diff
@@ -42,7 +40,12 @@ def rand_songs():
     size = int(size) if size else 10
     fromYear = int(fromYear) if fromYear else None
     toYear = int(toYear) if toYear else None
-    fid = uuid.UUID(musicFolderId) if musicFolderId else None
+    fid = None
+    if musicFolderId:
+        try:
+            fid = int(musicFolderId)
+        except ValueError:
+            raise ValueError("Invalid folder ID")
```
[the remark this answers is not in the document]
Comment (mvn23) [reply supysonic/api/albums_songs.py]: This is done to have a useful error message. Another option would be to change the existing ValueError and raise it.

File: supysonic/api/annotation.py [hunk 3]
```diff
@@ -77,6 +77,11 @@ def handle_star_request(func):
             terr = e
         try:
             func(Folder, eid)
+        except GenericError as e:
+            if e.message == "Invalid ID" and isinstance(terr, ObjectNotFound):
```
[the remark this answers is not in the document]
Comment (mvn23) [reply supysonic/api/annotation.py]: As the star request can now be called with either an int or an uuid in the id field, we need to check that neither exists before raising an invalid ID error. I agree that this is not the right way to do it and I will adjust the code accordingly.

**Comment by mvn23** [comment]
Small update, the last review comment has been addressed.
How would you prefer to deal with the other comments?

**Comment by spl0k** [comment]
Hi. Sorry for the late reply.
The other comments were more general remarks than calls for action. I'll merge this PR soon.

**Pull request merged**